### PR TITLE
CORGI-437: Fix many API bugs and reduce number of queries

### PR DESCRIPTION
**Pull request by juspence**
@RedHatProductSecurity/corgi-devs Part 1 of 2. This is some cleanup of our existing API and reduces the number of queries needed to load one page of results. Part 2 will be actual logic changes / new features to support OSC's requirements.

### CODE REVIEW

File: corgi/core/models.py [hunk 1]
```diff
@@ -403,17 +403,17 @@ def builds(self) -> QuerySet:
         return ProductComponentRelation.objects.none()
 
     @property
-    def coverage(self) -> int:
```
Comment (JimFuller-RedHat) [inline corgi/core/models.py:406]: I believe this function can be removed

File: corgi/api/serializers.py [hunk 2]
```diff
@@ -286,7 +286,7 @@ def get_sources(instance: Component) -> list[dict[str, str]]:
 
     @staticmethod
     def get_upstreams(instance: Component) -> list[dict[str, str]]:
-        return get_component_data_list(instance.get_upstreams_purls())
+        return get_component_data_list(instance.upstreams.values_list("purl", flat=True))
```
Comment (JimFuller-RedHat) [inline corgi/api/serializers.py:288]: very nice ;)

**Review by JimFuller-RedHat — COMMENTED** [review]
LGTM

File: corgi/core/models.py (hunk 1 above)
Comment (juspence) [reply corgi/core/models.py:406]: Ack, I'm going to merge here as soon as the pipeline passes (so I don't have to wait for the pipeline to run again). I'll push new commits for part 2 that remove this. Thanks for reviewing!

**Pull request merged**